### PR TITLE
fix: settings iconPrefixCls fail 

### PR DESCRIPTION
**Pull request by crazyair**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...
随便找个 demo 使用如下代码，查看 icon 没有设置成 `aaaa`
```tsx
import React from 'react';
import { ExclamationCircleFilled } from '@ant-design/icons';
import { Button, ConfigProvider, Modal } from 'antd';

const App: React.FC = () => {
  const [modal, contextHolder] = Modal.useModal();

  return (
    <ConfigProvider iconPrefixCls="aaaa" prefixCls="bbb">
      <ExclamationCircleFilled />
      <Button
        onClick={() => {
          modal.confirm({ title: 'aaa' });
        }}
      >
        aaaa
      </Button>
      <Button loading>loading</Button>
      {contextHolder}
    </ConfigProvider>
  );
};

export default App;
```

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     settings iconPrefixCls fail    |
| 🇨🇳 Chinese |    设置 `iconPrefixCls` 失效    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
补个用例？

File: components/config-provider/index.tsx [hunk 1]
```diff
@@ -514,9 +514,7 @@ const ProviderChildren: React.FC<ProviderChildrenProps> = (props) => {
   }
 
   if (iconPrefixCls || csp) {
-    childNode = (
-      <IconContext.Provider value={memoIconContextValue}>{childNode}</IconContext.Provider>
-    );
+    childNode = <IconProvider value={memoIconContextValue}>{childNode}</IconProvider>;
```
Comment (yoyo837) [inline components/config-provider/index.tsx:517]: 建议把 issue 注释在这里

Comment (crazyair) [reply components/config-provider/index.tsx:518]: done

**Comment by crazyair** [comment]
> 补个用例？

有用例，主要是开发模式跟测试模式不一样

https://github.com/ant-design/ant-design/blob/3d4e3ca089875ebbbb2e4ce4ad3befecf9cd927b/components/config-provider/__tests__/index.test.tsx#L79

**Comment by yoyo837** [comment]
已有的用例能调整后Cover住这个问题吗？不能的话就没办法了。

**Comment by crazyair** [comment]
> 已有的用例能调整后Cover住这个问题吗？不能的话就没办法了。

推荐是 Context 从包直接引入，避免环境是用 lib es 目录不一致导致获取不到上下文，单测的话，可能 test 跟 dev 不一样，所以就算写了单测也不能用来保障

**Review by MadCcc — APPROVED** [review]

**Pull request merged**